// Copyright 2017 MSO4SC - javier.carnero@atos.net
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//   http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

package slurm_framework;

message jobsettings {
	enum slurmCommandType {
		SRUN = 0;
		SBATCH = 1;
	}

	required slurmCommandType type = 1;
	required string command = 2;
	
	repeated string modules = 3;
	optional string partition = 4;
	optional int32 nodes = 5;
	optional bool exclusive = 6;
	optional int32 tasks = 7;
	optional int32 tasks_per_node = 8;
	optional int32 cpus_per_task = 9;
	optional int32 cores_per_node = 10;
	optional int32 mem = 11;
	optional int32 mem_per_cpu = 12;
	optional string max_time = 13;
}